### PR TITLE
Chef 13 compatibility

### PR DESCRIPTION
**Pull request by eldebrim**

### CODE REVIEW

**Comment by eldebrim** [comment]
[Chef 13: osl-jenkins](https://trello.com/c/ZsToblB9/868-chef-13-osl-jenkins)

File: recipes/default.rb [hunk 1]
```diff
@@ -43,3 +43,9 @@
 
 node.default['ssh_keys']['alfred'] = 'alfred'
 include_recipe 'ssh-keys'
+
+node.deep_fetch('osl-jenkins', 'gems').each do |g|
+  chef_gem g do # ~FC009
+    compile_time true
+  end
+end
```
Comment (ramereth) [inline recipes/default.rb]: This should be in the server recipe, not the default recipe.

File: recipes/ibmz_ci.rb [hunk 2]
```diff
@@ -33,16 +33,6 @@ class ::Chef::Recipe
...
-ruby_block 'Set jenkins username/password if needed' do
-  block do
-    if ::File.exist?('/var/lib/jenkins/config.xml') &&
-       ::File.foreach('/var/lib/jenkins/config.xml').grep(/GithubSecurityRealm/).any?
-      node.run_state[:jenkins_username] = secrets['git']['ibmz_ci']['user'] # ~FC001
-      node.run_state[:jenkins_password] = secrets['git']['ibmz_ci']['token'] # ~FC001
-    end
-  end
-end
-
```
Comment (ramereth) [inline recipes/ibmz_ci.rb]: Why was this removed?

File: recipes/plugins.rb [hunk 3]
```diff
@@ -28,10 +28,6 @@
 
 reload_file = ::File.join(Chef::Config[:file_cache_path], 'reload-jenkins')
 
-file reload_file do
-  action :nothing
-end
-
```
Comment (ramereth) [inline recipes/plugins.rb]: Why was this removed?

File: recipes/ibmz_ci.rb (hunk 2 above)
Comment (eldebrim) [reply recipes/ibmz_ci.rb]: This is a duplicate of the `ruby_block` in line 172.

File: recipes/plugins.rb (hunk 3 above)
Comment (eldebrim) [reply recipes/plugins.rb]: This has the same name as the `file` resource in line 48. They both have different actions so in Chef 12 the resource has a `:delete` action.

File: recipes/default.rb (hunk 1 above)
Comment (eldebrim) [reply recipes/default.rb]: Do you mean the master recipe?

File: recipes/ibmz_ci.rb (hunk 2 above)
Comment (ramereth) [reply recipes/ibmz_ci.rb]: I'd prefer if you removed the block at line 172 and keep the one at the top.

FYI: This is also happening in the powerci recipe.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -7,22 +7,22 @@
...
 
 ALL_PLATFORMS = [
   CENTOS_6_OPTS,
-  CENTOS_7_OPTS
+  CENTOS_7_OPTS,
 ]
 
 RSpec.configure do |config|
-  config.log_level = :fatal
+  config.log_level = :warn
```
Comment (ramereth) [inline spec/spec_helper.rb]: Let's not keep this set to ``warn`` long term for now.

File: recipes/master.rb [hunk 5]
```diff
@@ -64,8 +64,8 @@
     # .git-credentials file to work (needed for the cookbook_uploader recipe),
     # and the Jenkins git plugin recommends 1.8.x. So we use 1.8.5.5, the
     # latest 1.8.x.
-    node.set['git']['version'] = '1.8.5.5'
-    node.set['git']['checksum'] = '106b480e2b3ae8b02e5b6b099d7a4049' \
+    node.default['git']['version'] = '1.8.5.5'
+    node.default['git']['checksum'] = '106b480e2b3ae8b02e5b6b099d7a4049' \
```
Comment (ramereth) [inline recipes/master.rb]: Let's just make this one line.

File: recipes/master.rb [hunk 6]
```diff
@@ -116,3 +115,18 @@
   owner node['jenkins']['master']['user']
   group node['jenkins']['master']['group']
 end
+
+node.deep_fetch('osl-jenkins', 'gems').each do |g|
```
Comment (TravisWhitehead) [inline recipes/master.rb]: What is the purpose of `deep_fetch`?

Comment (ramereth) [reply recipes/master.rb]: If the node hash doesn't exist, don't error out.

**Comment by ramereth** [comment]
[Chef 13: Fix resource cloning with various roles](https://trello.com/c/YnEfltcl/923-chef-13-fix-resource-cloning-with-various-roles)

**Comment by eldebrim** [comment]
!bump major *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a major-level version bump to v2.0.0.  Have a nice day!
